cpp/src/arrow/dataset/arrow-dataset.pc.in

```diff
@@ -18,7 +18,14 @@
 libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
+so_version=@ARROW_SO_VERSION@
+abi_version=@ARROW_SO_VERSION@
+full_so_version=@ARROW_FULL_SO_VERSION@
+
 Name: Apache Arrow Datasets
 Description: Apache Arrow Datasets provides an API to read and write semantic datasets stored in different locations and formats.
 Version: @ARROW_VERSION@
 Requires: arrow
+
+Libs: -L${libdir} -larrow_dataset
+Cflags: -I${includedir} @ARROW_DEFINITION_FLAGS@
```

Comment:
We don't need `Cflags` because we don't need additional `Cflags` for Arrow Dataset.
We can get `-I${includddir} @ARROW_DEFINITION_FLAGS@` flags by `Requires: arrow`.